### DISCUSSION TITLE
How can i make a universal class for Prisma model with typescript?

### DISCUSSION THREAD

**Discussion by dimoff66** [prisma/prisma#5643] [Q&A]
I'm using Prisma with typescript
I created 2 models: Product and Category
I wanna make universal class for calling some queries

i do

```ts
export type DbTable = 'category' | 'product'
abstract class Dao {
  // define table
  abstract get dbTable (): DbTable 

  // getter for delegate
  private get dao() { return prisma[this.dbTable]  } 
  
  // common method
  public getAll() {
    return this.dao.findMany()
  }
}
```

But i get TypeScript error

_This expression is not callable.      
  Each member of the union type '(<T extends CategoryFindUniqueArgs, LocalRejectSettings 
= T["rejectOnNotFound"] extends RejectOnNotFound ? T["rejectOnNotFound"] : undefined>(args: SelectSubset<T, CategoryFindUniqueArgs>) => HasReject<...> extends 1 ? CheckSelect<...> : CheckSelect<...>) | (<T extends ProductFindUniqueArgs, LocalRejectSettings = T[...' has signatures, but none of those signatures are compatible with each other._

Is there are any way to reach the goal?

**Comment by ryands17** [comment]
@dimoff66 👋 

Currently it's not straightforward to implement this, but you can follow [this](https://github.com/prisma/prisma/issues/5273) request for workarounds and more updates as we are working on making this simpler by providing some sort of wrapper for this :)

**Reply by dimoff66** [reply]
Thank you, i already made my solution and it seems absolutely typesafe. I added it to that topic.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
